test/000_fullchain.test.ts

```diff
@@ -297,10 +298,125 @@ describe('Integration tests', function () {
-                    }
+                    expect(await iexecPoco.viewScore(workers[i].address)).to.be.equal(
+                        workerNumber == 1 ? 0 : 1,
+                    );
+                }
+            });
+        }
+        for (let workerNumber = 4; workerNumber < 6; workerNumber++) {
+            // Worker1 will contribute badly
+            it(`[7.${workerNumber - 3}] No sponsorship, no beneficiary, no callback, no BoT, up to ${workerNumber} workers with 1 bad actor`, async function () {
+                const volume = 1;
+                const workersAvailable = [worker1, worker2, worker3, worker4, worker5];
+                const { resultDigest: badResultDigest } = buildUtf8ResultAndDigest('bad-result');
+                const losingWorker = worker1;
+                const winningWorkers = workersAvailable.slice(1, workerNumber);
+                let workers = [{ signer: worker1, resultDigest: badResultDigest }];
+                for (const worker of winningWorkers) {
+                    workers.push({ signer: worker, resultDigest: resultDigest });
+                }
+                const acounts = [
+                    requester,
+                    scheduler,
+                    appProvider,
+                    datasetProvider,
+                    losingWorker,
+                    ...winningWorkers,
+                ];
+                // Create deal.
+                const orders = buildOrders({
+                    assets: ordersAssets,
+                    prices: ordersPrices,
+                    requester: requester.address,
+                    tag: standardDealTag,
+                    beneficiary: beneficiary.address,
+                    volume,
+                    trust: winningWorkers.length,
+                });
+
+                const { dealId, dealPrice, schedulerStakePerDeal } =
+                    await iexecWrapper.signAndMatchOrders(...orders.toArray());
+                const taskPrice = appPrice + datasetPrice + workerpoolPrice;
+                const schedulerStakePerTask = schedulerStakePerDeal / volume;
+                const workerRewardPerTask = await iexecWrapper.computeWorkerRewardPerTask(
+                    dealId,
+                    PocoMode.CLASSIC,
+                );
+                const schedulerRewardPerTask = workerpoolPrice - workerRewardPerTask;
+                // Check initial balances.
+                let accountsInitBalances = [
+                    {
+                        address: proxyAddress,
+                        frozen: (await iexecPoco.frozenOf(proxyAddress)).toNumber(),
+                    },
+                ];
+                for (const account of acounts) {
+                    let address = account.address;
+                    let frozen = (await iexecPoco.frozenOf(account.address)).toNumber();
+                    accountsInitBalances.push({ address, frozen });
+                }
```

Comment:
```suggestion
                for (const account of acounts) {
                    accountsInitialFrozens.push({
                        address: account.address,
                        frozen: (await iexecPoco.frozenOf(account.address)).toNumber(),
                    });
                }
```

Reply:
Also you can put all of the above in a function `const accountsInitialFrozens = getInitialFrozens(accounts)` and use it in the other test.

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/160/commits/83a8e64ddaffed58c766134199b94c27165f98be)